test/simplify_dyn_ops_test.cpp

```diff
@@ -0,0 +1,213 @@
+ * THE SOFTWARE.
+ */
+#include <migraphx/simplify_dyn_ops.hpp>
+#include <migraphx/split_single_dyn_dim.hpp>
+#include <migraphx/dead_code_elimination.hpp>
+#include <migraphx/program.hpp>
+#include <migraphx/pass_manager.hpp>
+#include <migraphx/make_op.hpp>
+#include <test.hpp>
+
+void run_pass(migraphx::module& m)
+{
+    migraphx::run_passes(m, {migraphx::simplify_dyn_ops{}, migraphx::dead_code_elimination{}});
+}
+
+TEST_CASE(static_broadcast)
+{
+    migraphx::module m0;
+    {
+        migraphx::shape s{migraphx::shape::float_type, {2, 4}};
+        auto input = m0.add_parameter("data", s);
+        migraphx::shape lit_s{migraphx::shape{migraphx::shape::float_type, {4}}};
+        auto literal_ins   = m0.add_literal(migraphx::literal{lit_s, {6, 5, 4, 3}});
+        auto broadcast_lit = m0.add_instruction(
+            migraphx::make_op("broadcast", {{"axis", 1}, {"out_lens", s.lens()}}), literal_ins);
+        auto add_ins = m0.add_instruction(migraphx::make_op("add"), input, broadcast_lit);
+        m0.add_return({add_ins});
+    }
+
+    migraphx::module m1;
+    {
+        migraphx::shape s{migraphx::shape::float_type, {2, 4}};
+        auto input = m1.add_parameter("data", s);
+        migraphx::shape lit_s{migraphx::shape{migraphx::shape::float_type, {4}}};
+        auto literal_ins = m1.add_literal(migraphx::literal{lit_s, {6, 5, 4, 3}});
+        auto broadcast_lit =
+            m1.add_instruction(migraphx::make_op("broadcast", {{"axis", 1}}), literal_ins, input);
+        auto add_ins = m1.add_instruction(migraphx::make_op("add"), input, broadcast_lit);
+        m1.add_return({add_ins});
+    }
+    run_pass(m1);
+
+    EXPECT(m0 == m1);
+}
+
+TEST_CASE(static_multibroadcast)
+{
+    migraphx::module m0;
+    {
+        migraphx::shape s{migraphx::shape::float_type, {2, 4}};
+        auto input = m0.add_parameter("data", s);
+        migraphx::shape lit_s{migraphx::shape{migraphx::shape::float_type, {1}, {0}}};
+        auto literal_ins   = m0.add_literal(migraphx::literal{lit_s, {6}});
+        auto broadcast_lit = m0.add_instruction(
+            migraphx::make_op("multibroadcast", {{"out_lens", s.lens()}}), literal_ins);
+        auto add_ins = m0.add_instruction(migraphx::make_op("add"), input, broadcast_lit);
+        m0.add_return({add_ins});
+    }
+
+    migraphx::module m1;
+    {
+        migraphx::shape s{migraphx::shape::float_type, {2, 4}};
+        auto input = m1.add_parameter("data", s);
+        migraphx::shape lit_s{migraphx::shape{migraphx::shape::float_type, {1}, {0}}};
+        auto literal_ins = m1.add_literal(migraphx::literal{lit_s, {6}});
+        auto broadcast_lit =
+            m1.add_instruction(migraphx::make_op("multibroadcast"), literal_ins, input);
+        auto add_ins = m1.add_instruction(migraphx::make_op("add"), input, broadcast_lit);
+        m1.add_return({add_ins});
+    }
+    run_pass(m1);
+
+    EXPECT(m0 == m1);
+}
+
+TEST_CASE(split_broadcast_match)
```

Comment:
I don't find `slice` in this test case.

Reply:
That's intended, this is mimicking simplifying the broadcasts after a split_single_dyn_dim

Reply:
so `split` was to refer to the fact that input is output from `split_single_dyn_dim` pass ?

Reply:
yes, I renamed the test to be more clear